src/components/data_search/DatasetSearchTable.jsx

```diff
@@ -76,12 +77,12 @@ export const DatasetSearchTable = (props) => {
       }
     ];
 
-    // do not apply search modifier if there is no searchTerm
-    if (searchTerm !== '') {
+    // do not apply search modifier if there is no search term
+    if (filters.search.length > 0) {
       const searchModifier = [
         {
           'multi_match': {
-            'query': searchTerm,
+            'query': filters.search[filters.search.length - 1],
```

Comment:
Why this change if searchTerm is being set in filterHandler?

Reply:
For reasons I don't completely understand `searchTerm` at this line reflects the previous `searchTerm` state (for example if your current input was `hello`, `searchTerm` would be `hell`)

Reply:
I'm wondering if something similar could be done everywhere searchTerm is used to eliminate the need for that variable at all - not sure if that would work right though